test/lint/lint-assertions.py

```diff
@@ -27,8 +27,9 @@ def main():
     # checks should be used over assert. See: src/util/check.h
     # src/rpc/server.cpp is excluded from this check since it's mostly meta-code.
     exit_code = git_grep([
-        "-nE",
-        r"\<(A|a)ss(ume|ert) *\(.*\);",
+        "--line-number",
+        "--extended-regexp",
+        r"\<(A|a)ss(ume|ert)\(",
```

Comment:
I think assuming no whitespace between a macro and the parenthesis is a regression, since that's allowed? (same for `BOOST_ASSERT`)

```cpp
#include <iostream>
#include <cstdlib>

#define Assume(x) do { if (!(x)) std::terminate(); } while(0)

int main() {
    Assume(true);  // Without space
    Assume (true); // With space

    std::cout << "Both assertions passed." << std::endl;
    return 0;
}
```

Reply:
Also, when I run this new `lint-assertions.py` on the old `rpc/blockchain.cpp` it still doesn't seem to catch the issue, I think because of the `\<` prefix? `r"(A|a)ss(ume|ert)\(",` works fine for me.

Reply:
> If you want to be more accurate you'll have to write a clang-tidy plugin, but I won't be doing this myself.

The issue doesn't seem very likely to happen, so I don't think we should spend too much time on this indeed, but since clang-format doesn't red the CI I feel like not removing two characters is a sensible approach to catch this. Either way is fine for me, so up to you.

> Keep in mind that macOS is not supported.

Ugh, sorry didn't consider that. I confirmed that the new test works fine on the old `src` code on my debian (grep GNU 3.8) machine.

```
./test/lint/lint-assertions.py
CHECK_NONFATAL(condition) or NONFATAL_UNREACHABLE should be used instead of assert for RPC code.
src/rpc/blockchain.cpp:804:    const auto& first_unpruned{*Assert(blockman.GetFirstBlock(*chain_tip, /*status_mask=*/BLOCK_HAVE_MASK, first_block))};
src/rpc/blockchain.cpp:811:    return Assert(first_unpruned.pprev)->nHeight;
```